### PR TITLE
Add pairwise function for 'gaussian' and 'concatenation' mode in NonLocal.

### PR DESCRIPTION
**Pull request by dreamerlin**
This pr adds pairwise function support for 'gaussian' and 'concatenation' mode in NonLocal. The Codes are referenced from [this](https://github.com/AlexHex7/Non-local_pytorch)

### CODE REVIEW

**Comment by dreamerlin** [comment]
Failed for `progress_bar`, not for `non_local`

File: mmcv/cnn/bricks/non_local.py [hunk 1]
```diff
@@ -43,12 +44,16 @@ def __init__(self,
         self.reduction = reduction
         self.use_scale = use_scale
         self.inter_channels = in_channels // reduction
+        if self.in_channels == 0:
+            self.in_channels = 1
```
Comment (xvjiarui) [inline mmcv/cnn/bricks/non_local.py:48]: `in_channels == 0` is not a valid input. We may remove this.

File: mmcv/cnn/bricks/non_local.py [hunk 2]
```diff
@@ -78,11 +71,38 @@ def __init__(self,
...
 
     def init_weights(self, std=0.01, zeros_init=True):
-        for m in [self.g, self.theta, self.phi]:
-            normal_init(m.conv, std=std)
+        if self.mode != 'gaussian':
+            for m in [self.g, self.theta, self.phi]:
+                normal_init(m.conv, std=std)
+        else:
+            normal_init(self.g.conv, std=std)
         if zeros_init:
```
Comment (xvjiarui) [inline mmcv/cnn/bricks/non_local.py:106]: We may also add test for `zero_init`.

File: mmcv/cnn/bricks/non_local.py (hunk 1 above)
Comment (dreamerlin) [reply mmcv/cnn/bricks/non_local.py]: A typo. It should be `inter_channels`.

File: mmcv/cnn/bricks/non_local.py (hunk 2 above)
Comment (dreamerlin) [reply mmcv/cnn/bricks/non_local.py:106]: Add it in `def test_nonlocal()`.

Comment (xvjiarui) [reply mmcv/cnn/bricks/non_local.py:106]: Strange. In coverage report 106-112 is not tested. Not sure about the reason.

Comment (dreamerlin) [reply mmcv/cnn/bricks/non_local.py:106]: Codes in 106-112 are tested now.

File: mmcv/cnn/bricks/non_local.py [hunk 3]
```diff
@@ -43,12 +44,16 @@ def __init__(self,
         self.reduction = reduction
         self.use_scale = use_scale
         self.inter_channels = in_channels // reduction
+        if self.inter_channels == 0:
+            self.inter_channels = 1
```
Comment (hellock) [inline mmcv/cnn/bricks/non_local.py:48]: `self.inter_channels = max(in_channels // reduction, 1)`

**Pull request merged**